port/unix/omrsock.c

```diff
@@ -76,8 +76,35 @@ omrsock_getaddrinfo_create_hints(struct OMRPortLibrary *portLibrary, omrsock_add
 
 int32_t
 omrsock_getaddrinfo(struct OMRPortLibrary *portLibrary, char *node, char *service, omrsock_addrinfo_t hints, omrsock_addrinfo_t result)
-{
-	return OMRPORT_ERROR_NOT_SUPPORTED_ON_THIS_PLATFORM;
+{	
+	omr_os_addrinfo *addrInfoResults = NULL;
+	omr_os_addrinfo *addrInfoHints = NULL;
+	int32_t count = 0;
+	
+	if (NULL == result) {
+		return OMRPORT_ERROR_INVALID_ARGUMENTS;
+	}
+	
+	memset(result, 0, sizeof(struct OMRAddrInfoNode));
+
+	if (NULL != hints) {
+		addrInfoHints = (omr_os_addrinfo *)hints->addrInfo;
+	}
+
+	if (0 != getaddrinfo(node, service, addrInfoHints, &addrInfoResults)) {
```

Comment:
Is it possible for getaddrinfo to return 0, but still set result to NULL (ie no matching results)?

Reply:
I believe it has to be 1 or more addrinfo node to be successful:
"Upon successful return, a pointer to a linked list of one or more addrinfo structures is returned through the res parameter."
"Given node and service, which identify an Internet host and a service, getaddrinfo() returns one or more addrinfo structures, each of which contains an Internet address that can be specified in a call to bind(2) or connect(2)."

Reply:
Makes sense, thanks 👍